tests/test_planning.py

```diff
@@ -283,17 +283,176 @@ def test_job_shop_problem():
 
 def test_refinements():
     
-    library = {'HLA': ['Go(Home,SFO)','Taxi(Home, SFO)'],
-    'steps': [['Taxi(Home, SFO)'],[]],
-    'precond': [['At(Home)'],['At(Home)']],
-    'effect': [['At(SFO)'],['At(SFO)'],['~At(Home)'],['~At(Home)']]}
+    library = {
+        'HLA': ['Go(Home,SFO)', 'Go(Home,SFO)', 'Drive(Home, SFOLongTermParking)', 'Shuttle(SFOLongTermParking, SFO)', 'Taxi(Home, SFO)'],
+        'steps': [['Drive(Home, SFOLongTermParking)', 'Shuttle(SFOLongTermParking, SFO)'], ['Taxi(Home, SFO)'], [], [], []],
+        'precond': [['At(Home)', 'Have(Car)'], ['At(Home)'], ['At(Home)', 'Have(Car)'], ['At(SFOLongTermParking)'], ['At(Home)']],
+        'effect': [['At(SFO)'], ['At(SFO)'], ['At(SFOLongTermParking)'], ['At(SFO)'], ['At(SFO)'], ['~At(Home)'], ['~At(Home)'], ['~At(Home)'], ['~At(SFOLongTermParking)'], ['~At(Home)']] }
+        
 
     go_SFO = HLA('Go(Home,SFO)', precond='At(Home)', effect='At(SFO) & ~At(Home)')
-    taxi_SFO = HLA('Go(Home,SFO)', precond='At(Home)', effect='At(SFO) & ~At(Home)')
+    taxi_SFO = HLA('Taxi(Home,SFO)', precond='At(Home)', effect='At(SFO) & ~At(Home)')
 
-    prob = Problem('At(Home)', 'At(SFO)', [go_SFO, taxi_SFO])
+    prob = Problem('At(Home) & Have(Car)', 'At(SFO)', [go_SFO, taxi_SFO])
 
     result = [i for i in Problem.refinements(go_SFO, prob, library)]
-    assert(len(result) == 1)
-    assert(result[0].name == 'Taxi')
-    assert(result[0].args == (expr('Home'), expr('SFO')))
+
+    for sequence in Problem.refinements(go_SFO, prob, library):
+        print ('ref',[(s.name, s.args) for s in sequence])
+
+    assert(len(result) == 2)
+    assert(result[0][0].name == 'Drive')
+    assert(result[0][0].args == (expr('Home'), expr('SFOLongTermParking')))
+    assert(result[0][1].name == 'Shuttle')
+    assert(result[0][1].args == (expr('SFOLongTermParking'), expr('SFO')))
+    assert(result[1][0].name == 'Taxi')
+    assert(result[1][0].args == expr('Home'), expr('SFO'))
+
+
+def test_convert_angelic_HLA():
+    """ 
+    Converts angelic HLA's into expressions that correspond to their actions
+    ~ : Delete (Not)
+    $+ : Possibly add (PosYes)
+    $-: Possibly delete (PosNo)
+    $$: Possibly add / delete (PosYesNo)
+    """
+    ang1 = Angelic_HLA('Test', precond = None, effect = '~A')
+    ang2 = Angelic_HLA('Test', precond = None, effect = '$+A')
+    ang3 = Angelic_HLA('Test', precond = None, effect = '$-A')
+    ang4 = Angelic_HLA('Test', precond = None, effect = '$$A')
+
+    assert(ang1.convert(ang1.effect) == [expr('NotA')])
+    assert(ang2.convert(ang2.effect) == [expr('PosYesA')])
+    assert(ang3.convert(ang3.effect) == [expr('PosNotA')])
+    assert(ang4.convert(ang4.effect) == [expr('PosYesNotA')])
+
+
+def test_is_primitive():
+    """
+    Tests if a plan is consisted out of primitive HLA's (angelic HLA's)
+    """
+    library = {
+        'HLA': ['Go(Home,SFO)', 'Go(Home,SFO)', 'Drive(Home, SFOLongTermParking)', 'Shuttle(SFOLongTermParking, SFO)', 'Taxi(Home, SFO)'],
```

Comment:
Since several test functions use this same library. Define the library once (as a global variable) at the beginining of the angelic_search tests and share that definition across the different tests.